File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int low = 0;
        int high = nums.length-1;
        int curr = 0;
        while(curr <= high){
            if(nums[curr] == 0){
                swap(nums, curr, low);
                low++;
                curr++;
            } else if(nums[curr] == 1){
                curr++;
            } else {
                swap(nums, curr, high);
                high--;
            }
        }
    }
    public void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: 0075-sort-colors/0075-sort-colors.go
func sortColors(nums []int)  {
    low := 0
    high := len(nums) -1
    curr := 0
    for curr <= high{
        if nums[curr] == 0{
            nums[low], nums[curr] = nums[curr], nums[low]
            low++
            curr++
        } else if nums[curr] == 2{
            nums[high], nums[curr] = nums[curr], nums[high]
            high--
        } else {
            curr++
        }
    }
}

// func swap(nums []int, i, j int) {
//     // temp := nums[i]
//     // nums[i] = nums[j]
//     // nums[j] = temp
//     nums[i], nums[j] = nums[j], nums[i]
// }